### ISSUE TITLE
ajax-select2: обратная дозаливка предка пишет голое имя, а <option> несёт ключ провайдера — поле сбрасывается

### ISSUE THREAD

**Issue by kalbac** [kalbac/woodev-plugin-framework#455]
Найдено при работе над #447 (заселение начального значения `ajax-select2` поля). Замерено тестом,
не предположение.

## Механизм

`buildSelectField()`/`applyEntries()` (`location-select-modes.js`) заводят `<option>` для
ajax-select2 с `value = entry.key` — внутренний ключ провайдера (`dadata:tatarstan` и т.п.):

```js
success( {
    results: ( entries || [] ).map( function( entry ) {
        return {
            id: entry.key,
            text: entry.record && entry.record.label ? entry.record.label : entry.label,
        };
    } ),
} );
```

Когда select2 вставляет выбранный результат в `<select>`, он создаёт `<option value="dadata:tatarstan">`
— значение поля становится КЛЮЧОМ, а не человекочитаемым именем.

Но обратная дозаливка предков (`location-cascade.js`, `backwardsFill()` → `writeSilently()`) пишет
СОВСЕМ ДРУГОЕ пространство значений — `fieldValueFor()` всегда возвращает голое ИМЯ компонента
(`record[level].name`, например `"Татарстан"`), никогда ключ:

```js
function writeSilently( entry, fieldId, value ) {
    ...
    var el = document.getElementById( fieldId );
    if ( el ) {
        el.value = value;   // "Татарстан" — а не "dadata:tatarstan"
    }
}
```

Для нативного `<select>` присвоение `.value` строке, которой нет среди `<option>`, тихо сбрасывает
выбор (`selectedIndex = -1`, `.value` становится `''`). Ровно это наблюдал оператор в #447,
симптом 4: «ввёл «Татарстан» в регионе, затем «Казань» в НП, выбрал Казань — поле региона
очистилось» — потому что `<option>` региона несёт `value="dadata:tatarstan"`, а `writeSilently()`
пишет `"Татарстан"`.

## Подтверждено тестом (не в этой PR — блокирующая находка, не фикс)

```
after key-based pick, select.value = "dadata:tatarstan"
after backwards-fill writeSilently(el.value = bare name), select.value = ""
```

## Почему это не в PR #450/#447

Фикс #447 (заселение НАЧАЛЬНОГО значения при построении поля — `initialValue`, взятое из
`input.value`) использует то же пространство значений, что и `writeSilently()` (голое имя), и
поэтому решает случай перезагрузки страницы. Но он не решает эту цепочку: любой ЖИВОЙ pick через
ajax-select2 создаёт option с value=KEY, а любая последующая обратная дозаливка пишет value=ИМЯ —
несовпадение остаётся.

Настоящий фикс требует согласованного решения на обеих сторонах:
- либо `location-select-modes.js` должен заводить `<option value>` в том же пространстве, что и
  `fieldValueFor()` (голое имя, не ключ) — но тогда `dataByKey`/`handleChange()` этого же файла
  нужно перевести на ключ по имени, что меняет поведение уже запиненное существующими тестами
  (`option.value === entry.key` в нескольких местах `location-select-modes.test.js`);
- либо `location-cascade.js::writeSilently()` должен знать про select2-поля и писать через
  `.val(key).trigger('change')`/пересобирать нужный `<option>`, а не голым `el.value =`.

Обе стороны меняют файл, которым в текущей карточке не владеет исполнитель (`location-cascade.js`
у параллельного воркера; изменение схемы ключей в `location-select-modes.js` — больше, чем просит
#447, и ломает запиненные тесты без явного запроса).

## Что сделать

Решить архитектурно: единое пространство значений для ajax-select2-полей между «что показывает
список» и «что пишет обратная дозаливка». Отдельная карточка, не довесок к #447/#450.

## Связано

- #447, #450 — карточки, в ходе которых это найдено.
- `woodev/shipping-method/assets/js/frontend/location-select-modes.js` (`applyEntries()`,
  `ensureSelect2()`'s `ajax.transport`).
- `woodev/shipping-method/assets/js/frontend/location-cascade.js` (`writeSilently()`,
  `fieldValueFor()`).

**Comment by kalbac** [comment]
## Подтверждено на риге 22.08.2026 — и симптом шире, чем описано в карточке

Оператор проверял `rig/s85-select2-verify` (= `main` + #454 + #456). После починки #447 расхождение пространств значений стало **видимым покупателю**: в поле «Town / City» отрисовано

```
dadata:0c5b2444-70a0-4932-980c-b4dc0d3f02b5
```

то есть ключ провайдера показан как название города.

### Почему это всплыло именно сейчас

Заселение из #447 берёт метку из самого значения (`seededOption.value = seededOption.textContent = initialValue`) — так делает и референсный плагин СДЭК, но там значение поля **и есть** человеческое название. У нас после живого выбора значение — ключ провайдера (`applyEntries()`: `option.value = entry.key`, `option.textContent = record.label`). Пока поле рендерилось пустым, ключ никто не видел. Теперь видно.

**#456 этого не создал — он это обнажил.** Само расхождение ровно то, что описано в этой карточке.

### Что это добавляет к постановке

К уже описанному «поле сбрасывается» добавляется второй симптом: **ключ провайдера как видимая метка**. Любая починка должна закрыть оба, то есть привести к одному пространству значений и отдельно решить, что показывать пользователю — а метка обязана браться из `record.label`, а не из значения.

### Чего в сессии НЕТ — проверено, чтобы не гонялись за призраком

```
usermeta.shipping_city              = Москва                    ← корректно
wp_woocommerce_sessions shipping_city = "" ×5 / "Moscow" ×3      ← ключа нет
woodev_customer_location            = полная запись, key = dadata:0c5b2444-…   ← by design
```

Ключ **не** лежит ни в `shipping_city` сессии WooCommerce, ни в user-meta. Он лежит только в нашей собственной сессионной записи, где ему и место. Каким путём он попадает в отрисованный `input` — **не установлено**; хранилище каскада живёт только в памяти и перезагрузку не переживает, так что путь пока не найден. Это надо выяснить до починки, иначе есть риск залечить симптом.

**Comment by kalbac** [comment]
Закрыто PR #461 (смержен в `main` 23.08.2026, `ac6a609`). Корень оказался общим для #455/#447/#460: слой держал два пространства значений и писал между ними без перевода — `location-select-modes.js` клал ключ провайдера в `<option value>` у `<select>`, несущего `name` самого поля. Теперь сабмитится значение поля, а идентичность записи переехала на `entry.key` через собственное событие select2 `select2:select`. Записи с пустым производным значением исключаются из выбираемых, а не получают фолбэк на ключ (находка критика Codex, доказанная вызовом рендерера).
